execution/state/state.go

```diff
@@ -130,7 +130,7 @@ func NewState(db dbm.DB) *State {
 }
 
 // Make genesis state from GenesisDoc and save to DB
-func MakeGenesisState(db dbm.DB, genesisDoc *genesis.GenesisDoc) (*State, error) {
+func MakeGenesisState(db dbm.DB, genesisDoc *genesis.GenesisDoc, restore DumpReader) (*State, error) {
```

Comment:
Can we split this into 3 functions: one private one with sig as currently, one as previously, and one with current sig that is public called `MakeGenesisStateFromDump` or similar.

Reply:
If split it in another way, please let me know what you think